kilkari/src/main/java/org/motechproject/nms/kilkari/domain/DeactivationSubscriptionRecord.java

```diff
@@ -0,0 +1,95 @@
+package org.motechproject.nms.kilkari.domain;
+
+import org.motechproject.mds.annotations.Entity;
+import org.motechproject.mds.annotations.Field;
+
+import javax.jdo.annotations.Column;
+import javax.validation.constraints.NotNull;
+
+@Entity(tableName = "nms_deactivation_subscription_audit_records")
+public class DeactivationSubscriptionRecord {
+
+    public static final int MAX_OUTCOME_LENGTH = 1000; // Includes the "Failure: " string
+
+    @Field
+    @NotNull
+    private String subscriptionId;
+
+    @Field
+    @NotNull
+    private Long subscriberId;
+
+    @Field
+    @NotNull
+    private SubscriptionOrigin subscriptionOrigin;
+
+    @Field
+    @NotNull
+    private Long msisdn;
+
+    @Field
+    @NotNull
+    private SubscriptionStatus preStatus;
+
+    public SubscriptionStatus getStatus() {
```

Comment:
@sripooja Ideally, you want have your getters and setters the same way as your variable name. Like getPreStatus or SetPreStatus. Otherwise, datanucleus ends up creating duplicate columns for the getters and setters without the prefix.